src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/LanguageClient/Extensions/TextDocumentExtensions.cs

```diff
@@ -10,12 +10,23 @@ namespace Microsoft.VisualStudio.Razor.LanguageClient.Extensions;
 internal static class TextDocumentExtensions
 {
     /// <summary>
-    /// Mutates the <see cref="TextDocumentIdentifier"/> by changing the <see cref="Uri"/>. Returns it again for convenience.
+    /// Returns a copy of the passed in <see cref="TextDocumentIdentifier"/> with the passed in <see cref="Uri"/>.
     /// </summary>
     public static TextDocumentIdentifier WithUri(this TextDocumentIdentifier textDocumentIdentifier, Uri uri)
     {
-        textDocumentIdentifier.Uri = uri;
```

Comment:
This isn't the first time this mutation has bit us, so I decided to just bite the bullet and remove it. We kick off Html and C# diagnostic requests at the same time, but its possible we were updating the uri to a html uri, then a C# uri, before the  instance was serialized. This just seems to be objectively bad.

Reply:
It was written on the walls that this day would come.